client/verta/verta/dataset/_dataset.py

```diff
@@ -53,3 +59,62 @@ def _path_component_to_repr_lines(path_component_msg):
             lines.append("    SHA-256 checksum: {}".format(path_component_msg.sha256))
 
         return lines
+
+    def _set_commit_and_blob_path(self, commit, blob_path):
+        """
+        Associate this blob with a commit and path to enable downloads.
+
+        Parameters
+        ----------
+        commit : :class:`verta._repository.commit.Commit`
+            Commit this blob was gotten from.
+        blob_path : str
+            Location of this blob within its Repository.
+
+        """
+        self._commit = commit
+        self._blob_path = blob_path
+
+    def download(self, component_path, filepath):
+        """
+        Downloads `component_path` from this dataset if ModelDB-managed versioning was enabled.
+
+        Parameters
+        ----------
+        component_path : str
+            Original path of the file in this dataset to download.
+        filepath : str
+            Filepath to download `component_path` to.
+
+        """
+        if self._commit is None and self._blob_path is None:
+            raise RuntimeError(
+                "this dataset cannot be used for downloads;"
+                " consider using `commit.get()` to obtain a download-capable dataset"
+                " if ModelDB-managed versioning was enabled"
+            )
+
+        # backend will return error if `component_path` not found/versioned
+        url = self._commit._get_url_for_artifact(self._blob_path, component_path, "GET").url
+
+        # stream download to avoid overwhelming memory
+        response = _utils.make_request("GET", url, self._commit._conn, stream=True)
+        try:
+            _utils.raise_for_http_error(response)
+
+            # decode responses that have Content-Encoding
```

Comment:
Check if this is compatible with the backend. For example, does it send gz automatically?

Reply:
Note that this request is being made to the artifact store, not to the ModelDB backend.

Regardless, I've decided to use `response.iter_content()` instead because I feel more comfortable with `requests`'s public API than poking into `urllib3`. S3 sends `application/octet-stream` anyway.